plugins/controlboard/src/ControlBoardDriverTrajectory.cpp

```diff
@@ -468,3 +471,134 @@ yarp::dev::TrajectoryType ConstSpeedTrajectoryGenerator::getTrajectoryType()
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+
+#if GAZEBO_MAJOR_VERSION >= 8
+    gazebo::physics::PhysicsEnginePtr physics = m_robot->GetWorld()->Physics();
+#else
+    gazebo::physics::PhysicsEnginePtr physics = m_robot->GetWorld()->GetPhysicsEngine();
+#endif
+
+    if (speed <= 0.0 || acceleration <= 0.0)
+    {
+        return false;
+    }
+
+    m_x0 = current_pos;
+    m_xf = final_pos;
+    m_v0 = m_computed_reference_velocity; // last computed velocity from previous trajectory
+
+    if (m_xf > m_joint_max) m_xf = m_joint_max;
+    else if (m_xf < m_joint_min) m_xf = m_joint_min;
+
+    double sign = (m_xf >= m_x0) ? 1.0 : -1.0;
+    m_speed = sign * speed;
+    m_acceleration = sign * acceleration;
+
+    // hypothetic distance traveled using a double-ramp (i.e. triangular) trajectory that reaches m_speed
+    double deltaTriMax = 0.5 * (2 * std::pow(m_speed, 2) - std::pow(m_v0, 2)) / m_acceleration;
+    double deltaTotal = m_xf - m_x0;
+
+    if (std::abs(deltaTotal) > std::abs(deltaTriMax))
+    {
+        // trapezoidal profile
+        double deltaRamp1 = 0.5 * (std::pow(m_speed, 2) - std::pow(m_v0, 2)) / m_acceleration;
+        double deltaRamp2 = 0.5 * std::pow(m_speed, 2) / m_acceleration;
+        m_ta = (m_speed - m_v0) / m_acceleration;
+        m_tb = m_ta + (deltaTotal - deltaRamp1 - deltaRamp2) / m_speed;
+        m_tf = m_tb + (m_speed / m_acceleration);
+    }
+    else
+    {
+        // triangular profile
+        double peakVelocity = sign * std::sqrt(0.5 * (2 * m_acceleration * deltaTotal + std::pow(m_v0, 2)));
+        m_ta = (peakVelocity - m_v0) / m_acceleration;
+        m_tb = m_ta;
+        m_tf = m_ta + (peakVelocity / m_acceleration);
+    }
+
+    m_computed_reference = current_pos;
+    m_controllerPeriod = physics->GetUpdatePeriod() * 1000.0; // milliseconds
+    m_tick = 0;
+    m_trajectory_complete = false;
+
+    return true;
+}
+
+bool TrapezoidalSpeedTrajectoryGenerator::abortTrajectory(double limit)
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    m_trajectory_complete = true;
+    m_computed_reference_velocity = 0.0;
+    return true;
+}
+
+double TrapezoidalSpeedTrajectoryGenerator::computeTrajectory()
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    m_computed_reference += p_computeTrajectoryStep();
+    return m_computed_reference;
+}
+
+double TrapezoidalSpeedTrajectoryGenerator::computeTrajectoryStep()
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    return p_computeTrajectoryStep();
+}
+
+double TrapezoidalSpeedTrajectoryGenerator::p_computeTrajectoryStep()
+{
+    if (m_trajectory_complete)
+    {
+        return 0;
+    }
+
+    double period = m_controllerPeriod / 1000.0; // in seconds
+    double instant = m_tick * period; // current time since start (in seconds)
+    double step = 0;
+
+    if (instant <= m_ta) // first ramp
+    {
+        step = 0.5 * m_acceleration * std::pow(period, 2.0) * (2 * m_tick + 1) + m_v0 * period;
```

Comment:
Nitpick: I think (not 100% sure to be honest) that  `std::pow(period, 2)` should use the simple period*period while specifying the exponent as double will take another more expensive overload.